src/python/pants/build_graph/target.py

```diff
@@ -473,7 +473,10 @@ def transitive_invalidation_hash(self, fingerprint_strategy=None, depth=0):
       def dep_hash_iter():
         for dep in self.dependencies:
           try:
-            dep_hash = dep.transitive_invalidation_hash(fingerprint_strategy, depth=depth+1)
+            if direct:
+              dep_hash = dep.invalidation_hash(fingerprint_strategy)
```

Comment:
No it's not, at least not in current implementation.

transitive_invalidation_hash method hashes dependencies recursively, "depth" parameter here does not mean the wanted depth of invalidation hash. It is only used to track depth of recursion and avoid hitting python stack limit.